components/eamxx/src/python/libpyscream/pyfield.hpp

```diff
@@ -59,8 +69,9 @@ struct PyField {
 
     // NOTE: you MUST set the parent handle, or else you won't have view semantic
     auto data = f.get_internal_view_data_unsafe<void,Host>();
-    auto this_obj = pybind11::cast(this);
-    return pybind11::array(dt,shape,strides,data,pybind11::handle(this_obj));
+    auto this_obj = nb::cast(this);
+    // pybind11::array(dt,shape,strides,data,pybind11::handle(this_obj));
+    return nb::ndarray<X,Y,Z>(data, shape_t, shape, nb::handle(this_obj));
```

Comment:
Related to template arg comments: I think it's fine (for now) to hard-code X to be `nb::numpy`. We can decide how to allow getting different backends later, if needed. The other two args can be removed altogether, I think.